tests-e2e/core/gitopsDeployment_test.go

```diff
@@ -0,0 +1,534 @@
+						Namespace: fixture.GitOpsServiceE2ENamespace,
+						Name:      name,
+						Status:    managedgitopsv1alpha1.SyncStatusCodeSynced,
+						Health: &managedgitopsv1alpha1.HealthStatus{
+							Status: managedgitopsv1alpha1.HeathStatusCodeHealthy,
+						},
+					},
+					{
+						Group:     "route.openshift.io",
+						Version:   "v1",
+						Kind:      "Route",
+						Namespace: fixture.GitOpsServiceE2ENamespace,
+						Name:      name,
+						Status:    managedgitopsv1alpha1.SyncStatusCodeSynced,
+						Health: &managedgitopsv1alpha1.HealthStatus{
+							Status:  managedgitopsv1alpha1.HeathStatusCodeHealthy,
+							Message: "Route is healthy",
+						},
+					},
+					{
+						Group:     "",
+						Version:   "v1",
+						Kind:      "Service",
+						Namespace: fixture.GitOpsServiceE2ENamespace,
+						Name:      name,
+						Status:    managedgitopsv1alpha1.SyncStatusCodeSynced,
+						Health: &managedgitopsv1alpha1.HealthStatus{
+							Status: managedgitopsv1alpha1.HeathStatusCodeHealthy,
+						},
+					},
+				}
+			}
+			expectedResourceStatusList := []managedgitopsv1alpha1.ResourceStatus{
+				{
+					Group:     "",
+					Version:   "v1",
+					Kind:      "ConfigMap",
+					Namespace: fixture.GitOpsServiceE2ENamespace,
+					Name:      "environment-config-map",
+					Status:    managedgitopsv1alpha1.SyncStatusCodeSynced,
+				},
+			}
+			expectedResourceStatusList = append(expectedResourceStatusList, getResourceStatusList("component-a")...)
+			expectedResourceStatusList = append(expectedResourceStatusList, getResourceStatusList("component-b")...)
+
+			Eventually(gitOpsDeploymentResource, "2m", "1s").Should(
+				SatisfyAll(
+					gitopsDeplFixture.HaveSyncStatusCode(managedgitopsv1alpha1.SyncStatusCodeSynced),
+					gitopsDeplFixture.HaveHealthStatusCode(managedgitopsv1alpha1.HeathStatusCodeHealthy),
+					gitopsDeplFixture.HaveResources(expectedResourceStatusList),
+				),
+			)
+			By("delete the GitOpsDeployment resource")
+
+			err = k8s.Delete(&gitOpsDeploymentResource)
+			Expect(err).To(Succeed())
+
+			err = k8s.Get(&gitOpsDeploymentResource)
+			Expect(err).ToNot(Succeed())
+
+			for _, resourceValue := range expectedResourceStatusList {
+				ns := typed.NamespacedName{
+					Namespace: resourceValue.Namespace,
+					Name:      resourceValue.Name,
+				}
+				err := k8sClient.Get(ctx, ns, &gitOpsDeploymentResource)
+				Expect(err).ToNot(Succeed())
+			}
+
+		})
+
+		It("Checks for failure of deployment when an invalid input is provided", func() {
+
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+
+			By("ensuring GitOpsDeployment Fails to create and deploy application when an invalid field input is passed")
+			gitOpsDeploymentResource := buildGitOpsDeploymentResource("",
+				"abc", "path/path/path",
+				managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated)
+
+			err := k8s.Create(&gitOpsDeploymentResource)
+			Expect(err).ToNot(Succeed())
+
+			Eventually(gitOpsDeploymentResource, "2m", "1s").ShouldNot(
```

Comment:
I believe this should be an `Eventually` followed by a `Consistently`.

The Eventually should wait for it to have the target expected state (`unhealthy`/`unsynced`, I guess), then Consistently should ensure it stays in that state for 60 seconds (checking once per second).